lib/store.js

```diff
@@ -20,14 +25,23 @@ const initialState = Immutable.fromJS({
 export const store = createStore(reducer);
 
 // Actions
-
 export function set(path, value) {
-  dispatch({type: 'SET_IN', path, value});
+  store.dispatch({type: 'SET_IN', path, value});
 }
 
 // Reducer
 function reducer(state = initialState, action) {
+  console.log("action:", action);
   switch (action.type) {
+    case 'SEARCH_START':   //search has been started
+      state = state.setIn( ['search', 'q'] , action.q);
+      state = state.setIn( ['search','results'] , Immutable.fromJS([]));
+      return state.setIn( ['search','isActve'], true);
+    case 'SEARCH_RESULT':   //one or more search result has come in
+      state = state.setIn( ['search', 'q'], action.q );
+      state = state.setIn( ['search','results'] , Immutable.fromJS(action.results));
+      state = state.setIn( ['search','isActive'] , false);
+      return state;
```

Comment:
looks fine, my approach would probably be something like this (when I think more about it):
```javascript
return state.mergeDeep(Immutable.fromJS({search: {
  q: action.q,
  results: action.results,
  isActive: false
}});
```
(haven't tested it)